### ISSUE TITLE
Add support for ignoring blocks of the readme

### ISSUE THREAD

**Issue by jthegedus** [sindresorhus/awesome-lint#54]
[<img alt="Issuehunt badges" src="https://issuehunt.io/r/sindresorhus/awesome-lint/issues/54/badge.svg" />](https://issuehunt.io/r/sindresorhus/awesome-lint/issues/54)

I am using [all-contributors-cli](https://github.com/jfmengels/all-contributors-cli) which generates the contributors section oy my readme. The generated code throws this error:

```shell
Titles should use ' as a quote               link-title-style
```

I could move the contributors to it's own file, but I would like to keep it front and center in the readme.

If we could get ignores (preferably range ignores) `<!-- awesome-lint-ignore-start -->` like [prettier](https://prettier.io/docs/en/ignore.html#range-ignore) then I could use `awesome-lint` alongside `all-contributors`.

> There is a $40.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/awesome-lint/issues/54).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/awesome-lint/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

**Comment by sindresorhus** [comment]
We could add https://github.com/remarkjs/remark-message-control

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $40.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/64659785/issues/54) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/64659785/issues/54).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/users/44827199/repositories) on IssueHunt to raise funds.

**Comment by chinesedfan** [comment]
Good news is that we have already used `remark-lint`, which introduced `remark-message-control`.

Bad news is that our ruleIds, i.e. awesome/badge, contain a slash, which is invalid as parameter. See [mdast-comment-marker](https://github.com/syntax-tree/mdast-comment-marker/blob/master/index.js#L7). So we can't specify ruleIds. Followings do not work.

```
<!--lint disable awesome/list-item-->
- [foo](https://foo.com) - an invalid description.
```

@sindresorhus @transitive-bullshit What if we rename ruleIds i.e. `awesome-badge`? It may affect something.

**Comment by transitive-bullshit** [comment]
I would rather fix remark's ability to recognize eslint-style rule names and keep our current rule naming convention than change our naming convention due to an unnecessary constraint within `remark`.

@wooorm what are your thoughts on allowing this style of lint rule naming?

**Comment by wooorm** [comment]
Message-control and comment-marker conform to HTML, it's like attributes, which I think makes sense for markdown.
Note that we also use colons throughout the ecosystem. To differentiate between the 'source' (in this case awesome) and the ruled.
That's the reasoning here, but feel free to open an issue!

**Comment by chinesedfan** [comment]
@wooorm Please check my PR, https://github.com/syntax-tree/mdast-comment-marker/pull/5 and another question https://github.com/unifiedjs/unified-message-control/issues/4.

**Comment by sindresorhus** [comment]
I'm happy to switch to colons. The rule naming doesn't matter much and following the eslint-style rule naming is not import. @transitive-bullshit ?

**Comment by transitive-bullshit** [comment]
Sounds good to me.
